assign_rights/assemble.py

```diff
@@ -6,9 +8,11 @@
 class RightsAssembler(object):
     """docstring for RightsCalculator"""
 
-    def retrieve_rights(self):
-        """docstring for retrieve_rights"""
-    pass
+    def retrieve_rights(self, identifier):
+        """Retrieves a rights shell whose rights_id matches an identifier from
+        a post request.
+        """
+        return RightsShell.objects.get(pk=identifier)
```

Comment:
After refactoring, this is now a one-line function, so I think there are two possible paths here:
1. move the line to the `run` method.
2. move the `for` loop currently in the `run` method into this function.

Thinking about it, my preference would be for the latter, because you need to know you have all the rights shells before processing, so it's more efficient if you make sure those all exist first before trying to calculate dates and/or assemble the statement.

So this function would be:
```suggestion
    def retrieve_rights(self, rights_ids):
        """Retrieves a rights shell whose rights_id matches an identifier from
        a post request.
        """
        rights_shells = []
        for ident in rights_ids:
            rights_shells.append(RightsShell.objects.get(pk=ident))
        return rights_shells
```

Reply:
Ok, latter option makes more sense to me as well. I just spent 15 minutes thinking about the best way to calculate dates with the current for loop and it wasn't going to be great.